lib/scenario-editor/utils/valhalla.js

```diff
@@ -198,26 +209,69 @@ export async function getSegment (
  *
  * Example URL: https://graphhopper.com/api/1/route?point=49.932707,11.588051&point=50.3404,11.64705&vehicle=car&debug=true&&type=json
  */
-export function routeWithGraphHopper (points: Array<LatLng>, avoidMotorways?: boolean): ?Promise<GraphHopperResponse> {
+export function routeWithGraphHopper (
+  points: Array<LatLng>,
+  avoidMotorways?: boolean,
+  snapToOption: string
+): ?Promise<GraphHopperResponse> {
   if (points.length < 2) {
     console.warn('need at least two points to route with graphhopper', points)
     return null
   }
-  if (!process.env.GRAPH_HOPPER_KEY) {
-    throw new Error('GRAPH_HOPPER_KEY not set')
-  }
+
+  let graphHopperUrl = ''
+  let graphHopperKey = ''
+
   // Use custom url if it exists, otherwise default to the hosted service.
-  const GRAPH_HOPPER_URL = process.env.GRAPH_HOPPER_URL || 'https://graphhopper.com/api/1/'
+  if (process.env.GRAPH_HOPPER_ALTERNATES) {
+    // $FlowFixMe This is a bit of a hack and not how env variables are supposed to work, but the yaml loader supports it.
+    const alternates: Array<GraphHopperAlternateServer> = process.env.GRAPH_HOPPER_ALTERNATES
+    alternates.forEach(alternative => {
+      const { BBOX, ROUTING_TYPE, URL, KEY } = alternative
+      if (BBOX.length !== 4) {
+        console.warn('Invalid BBOX for GRAPH_HOPPER_ALTERNATIVE')
+        return
+      }
+      if (!URL) {
+        console.warn('No URL provided for alternative graphhopper server. Using open testing environment.')
+      }
+
+      if (arePointsValid(points, BBOX)) {
+        if (snapToOption === 'RAIL') {
+          if (ROUTING_TYPE === 'RAIL' && URL) {
+            graphHopperUrl = URL
+          } else { // no URL or need default value so use openrailrouting test environment
+            graphHopperUrl = 'https://routing.openrailrouting.org/'
+          }
+        } else if (snapToOption === 'STREET') {
+          if (ROUTING_TYPE === 'STREET' && KEY && URL) {
+            graphHopperUrl = URL
+            graphHopperKey = KEY
+          } else { // no URL or need default value so use graphhopper test environment
+            graphHopperUrl = 'https://graphhopper.com/api/1/'
+            // include key here??? this will throw 401 i believe
```

Comment:
You deleted the code that makes it required to have a graphhopper key! That's a required param